### DISCUSSION TITLE
The ability to apply middleware to rewritten api requests

### DISCUSSION THREAD

**Discussion by abeluck** [vercel/next.js#19432] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

We're using the API rewriting feature of next.js to rewrite calls to a backend system. This system expects all API calls to be authorized with a Bearer token in the Authorization header. However our next.js frontend is using a session cookie to hold the token.

Given the current feature set we cannot proxy calls to the backend system unless we manually write the rewrites.

## Describe the solution you'd like

The ability to rewrite parts of the incoming/outgoing request as its re-written to the remote. This would probably take the form of some sort of middleware function.

## Describe alternatives you've considered

We're unable to change the backend system to extract the token from a cookie rather than the authorization header.

We've also considered manually proxying the calls using something like the technique [described here](https://github.com/vercel/next.js/pull/17557#issuecomment-708517053), but this requires a bunch of extra dependencies, something our environment is sensitive to.

## Additional context

This is somewhat related to https://github.com/vercel/next.js/pull/17557

**Comment by code-press** [comment]
Have you tried this? https://github.com/oslabs-beta/connext-js
